applications/opibuilder/opibuilder-plugins/org.csstudio.opibuilder/src/org/csstudio/opibuilder/scriptUtil/PVUtil.java

```diff
@@ -87,211 +91,243 @@ public void partDeactivated(EditPart arg0) {
+        for (double d : dblArray) {
             longArray[i++] = (long) d;
         }
         return longArray;
     }
 
-    /**Get the size of the pv's value
-     * @param pv the pv.
-     * @return Array length of the pv value. <code>1</code> for scalars. */
-    public final static double getSize(IPV pv){
+    /**
+     * Get the size of the pv's value
+     *
+     * @param pv
+     *            the pv.
+     * @return Array length of the pv value. <code>1</code> for scalars.
+     */
+    public final static double getSize(IPV pv) {
         return VTypeHelper.getSize(checkPVValue(pv));
     }
 
-
-      /**
-     * Converts the given pv's value into a string representation. For string values,
-     * returns the value. For numeric (double and long) values, returns a
-     * non-localized string representation. Double values use a point as the
-     * decimal separator. For other types of values, the value's
-     * {@link IValue#format()} method is called and its result returned.
+    /**
+     * Converts the given pv's value into a string representation. For string values, returns the value. For numeric (double and
+     * long) values, returns a non-localized string representation. Double values use a point as the decimal separator. For other
+     * types of values, the value's {@link IValue#format()} method is called and its result returned.
      *
      * @param pv
      *            the pv.
      * @return a string representation of the value.
      */
-    public final static String getString(IPV pv){
+    public final static String getString(IPV pv) {
         return VTypeHelper.getString(checkPVValue(pv));
     }
 
-    /**Get the full info from the pv in this format
-     * <pre>timestamp value severity, status</pre>
+    /**
+     * Get the full info from the pv in this format
+     *
+     * <pre>
+     * timestamp value severity, status
+     * </pre>
+     *
      * @param pv
      * @return the full info string
      */
-    public final static String getFullString(IPV pv){
+    public final static String getFullString(IPV pv) {
         return checkPVValue(pv).toString();
     }
 
+    /**
+     * Get the timestamp string of the pv
+     *
+     * @param pv
+     *            the pv
+     * @return the timestamp in string.
+     */
+    public final static String getTimeString(IPV pv) {
+        Instant time = VTypeHelper.getTimestamp(checkPVValue(pv));
+        if (time != null)
+            return timeFormat.format(time);
+        return ""; //$NON-NLS-1$
+    }
 
-    /**Get the timestamp string of the pv
-     * @param pv the pv
+    /***
+     * Get the timestamp string of the pv
+     *
+     * @param pv
+     *            the Pv
+     * @param formatPattern
+     *            format pattern in the form of {@link DateTimeFormatter}
      * @return the timestamp in string.
      */
-    public final static String getTimeString(IPV pv){
+
+    public final static String getTimeString(IPV pv, String formatPattern) {
```

Comment:
Yes, the formatting seems to change the whole file, but this is the only change.